addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/sns/NotificationComponent.java

```diff
@@ -0,0 +1,147 @@
+
+    protected AmazonSNS amazonSNS;
+
+    protected URI endpointURL;
+
+    protected final Map<String, TopicDescriptor> topics = new HashMap<>();
+
+    public static final String TOPICS_XP = "topics";
+
+    @Override
+    public void registerContribution(Object contribution, String xp, ComponentInstance component) {
+        if (TOPICS_XP.equals(xp)) {
+            TopicDescriptor topic = (TopicDescriptor) contribution;
+            topics.put(topic.getType(), topic);
+        } else {
+            super.registerContribution(contribution, xp, component);
+        }
+    }
+
+    @Override
+    public void start(ComponentContext context) {
+        super.start(context);
+        topics.forEach((name, topic) -> {
+            CreateTopicResult result = createTopic(topic.name);
+            topic.setArn(result.getTopicArn());
+
+            try {
+                subscribe(result.getTopicArn(), getURL(topic.getPath()));
+            } catch (URISyntaxException e) {
+                log.error(e);
+            }
+        });
+    }
+
+    @Override
+    public void stop(ComponentContext context) throws InterruptedException {
+        super.stop(context);
+        amazonSNS = null;
+    }
+
+
+    @Override
+    public AmazonSNS getClient() {
+        return client();
+    }
+
+    @Override
+    public CreateTopicResult createTopic(String name) {
+        final CreateTopicRequest request = new CreateTopicRequest(name);
+        return client().createTopic(request);
+    }
+
+    @Override
+    public void removeTopic(String name) {
+        if (topics.containsKey(name)) {
+            client().deleteTopic(topics.get(name).arn);
+        } else {
+            log.info("An attempt to remove non-existent topic with name " + name);
+        }
+    }
+
+    @Override
+    public void subscribe(String arn, URI uri) {
+        SubscribeRequest https = new SubscribeRequest(arn, uri.getScheme(), uri.toString());
+        client().subscribe(https);
+    }
+
+    @Override
+    public String getArnFor(String topicType) {
+        if (!topics.containsKey(topicType)) {
+            return null;
+        }
+        return topics.get(topicType).getArn();
+    }
+
+    @Override
+    public int getApplicationStartedOrder() {
+        return 500;
+    }
+
+    protected AmazonSNS client() {
+        if (amazonSNS != null) {
+            return amazonSNS;
+        }
+
+        synchronized (this) {
+            amazonSNS = AmazonSNSClientBuilder.standard()
+                    .withCredentials(AWSHelper.getInstance().getCredentialsProvider())
+                    .withRegion(AWSHelper.getInstance().getRegion())
+                    .build();
+            return amazonSNS;
+        }
+    }
+
+    protected URI getURL(String path) throws URISyntaxException {
+        if (endpointURL == null) {
+            String host = (String) Framework.getProperties().getOrDefault("nuxeo.url", "http://0.0.0.0:8080");
```

Comment:
Was protocol needed after all?

Reply:
You re-used `nuxeo.url` -  not keen on that assumption

Reply:
`nuxeo.url` is the default property for getting public host across most of our addons.

Reply:
As long as you are sure.